manager/integration/tests/backupstore.py

```diff
@@ -17,14 +17,104 @@
+    yield
+    reset_backupstore_setting(client)
+
+
+@pytest.fixture
+def backupstore_nfs(client):
+    set_backupstore_nfs(client)
+    yield
+    reset_backupstore_setting(client)
+
+
+@pytest.fixture(params=["s3", "nfs"])
+def set_random_backupstore(request, client):
+    if request.param == "s3":
+        set_backupstore_s3(client)
+    elif request.param == "nfs":
+        set_backupstore_nfs(client)
+        mount_nfs_backupstore(client)
+
+    yield
+    backupstore_cleanup(client)
+    reset_backupstore_setting(client)
+
+    if request.param == "nfs":
+        umount_nfs_backupstore(client)
+
+
+def reset_backupstore_setting(client):
+    backup_target_setting = client.by_id_setting(SETTING_BACKUP_TARGET)
+    client.update(backup_target_setting, value="")
+    backup_target_credential_setting = client.by_id_setting(
+        SETTING_BACKUP_TARGET_CREDENTIAL_SECRET)
+    client.update(backup_target_credential_setting, value="")
+
+
+def set_backupstore_s3(client):
+    backup_target_setting = client.by_id_setting(SETTING_BACKUP_TARGET)
+    backupstores = get_backupstore_url()
+    for backupstore in backupstores:
+        if is_backupTarget_s3(backupstore):
+            backupsettings = backupstore.split("$")
+            backup_target_setting = client.update(backup_target_setting,
+                                                  value=backupsettings[0])
+            assert backup_target_setting.value == backupsettings[0]
+
+            backup_target_credential_setting = client.by_id_setting(
+                SETTING_BACKUP_TARGET_CREDENTIAL_SECRET)
+            backup_target_credential_setting = \
+                client.update(backup_target_credential_setting,
+                              value=backupsettings[1])
+            assert backup_target_credential_setting.value == backupsettings[1]
+            break
+
+
+def set_backupstore_nfs(client):
+    backup_target_setting = client.by_id_setting(SETTING_BACKUP_TARGET)
+    backupstores = get_backupstore_url()
+    for backupstore in backupstores:
+        if is_backupTarget_nfs(backupstore):
+            backup_target_setting = client.update(backup_target_setting,
+                                                  value=backupstore)
+            assert backup_target_setting.value == backupstore
+            backup_target_credential_setting = client.by_id_setting(
+                SETTING_BACKUP_TARGET_CREDENTIAL_SECRET)
+            backup_target_credential_setting = \
+                client.update(backup_target_credential_setting, value="")
+            assert backup_target_credential_setting.value == ""
+            break
+
+
+def mount_nfs_backupstore(client, mount_path="/mnt/nfs"):
+    cmd = ["mkdir", "-p", mount_path]
+    subprocess.check_output(cmd)
+    nfs_backuptarget = client.by_id_setting(SETTING_BACKUP_TARGET).value
+    nfs_url = urlparse(nfs_backuptarget).netloc + \
+        urlparse(nfs_backuptarget).path
+    cmd = ["mount", "-t", "nfs4", nfs_url, mount_path]
```

Comment:
Can you change the mount to use vers=4.1 ?
nfs v4.1 has better performance and better state managment, we don't really want to support v4 on the server.
The client will negotiate down to v4 if the server doesn't support 4.1, your ganesha test server supports 4.1 and 4.2
We might even consider going with 4.2 by the default but the change between 4.2 and 4.1 is smaller then the big difference between 4.1 and 4.0

Reply:
Changed NFS version to use 4.2 by default.